README.md

```diff
@@ -354,20 +354,59 @@ for hiredis-cluster as well.
 
 ### Connecting
 
-The function `redisClusterAsyncConnect` can be used to
-establish a set of non-blocking connections to a Redis cluster.
-It returns a pointer to the newly created `redisClusterAsyncContext` struct.
-The `err` field should be checked after creation
-to see if there were errors creating the asynchronous cluster context.
+There are two alternative ways to initiate a cluster client which also determines
+how the client behaves during the initial connect.
+
+The first alternative is to use the function `redisClusterAsyncConnect`, which initially
+connects to the cluster in a blocking fashion and waits for the slotmap before returning.
+Any command sent by the user thereafter will create a new non-blocking connection,
+unless a non-blocking connection already exists to the destination.
+The function returns a pointer to a newly created `redisClusterAsyncContext` struct and
+its `err` field should be checked to make sure the initial slotmap update was successful.
 
 ```c
+// Insufficient error handling for brevity.
 redisClusterAsyncContext *acc = redisClusterAsyncConnect("127.0.0.1:6379", HIRCLUSTER_FLAG_NULL);
 if (acc->err) {
-    printf("Error: %s\n", acc->errstr);
-    // handle error
+    printf("error: %s\n", acc->errstr);
+    exit(1);
+}
+
+// Attach an event engine. In this example we use libevent.
+struct event_base *base = event_base_new();
+redisClusterLibeventAttach(acc, base);
+```
+
+The second alternative is to use `redisClusterAsyncContextInit` and `redisClusterAsyncConnect2`
+which avoids the initial blocking connect. This connection alternative requires an attached
+event engine when `redisClusterAsyncConnect2` is called, but the connect and the initial
+slotmap update is done in a non-blocking fashion.
+
+This means that commands sent directly after `redisClusterAsyncConnect2` may fail
+because the initial slotmap has not yet been retrieved and the client dont know which
+cluster node to send the command to. You may use the [eventCallback](#events-per-cluster-context)
+to be notified when the slotmap is updated and the client is ready to accept commands.
+An crude example of using the eventCallback can be found in [this testcase](tests/ct_async.c).
+
+```c
+// Insufficient error handling for brevity.
+redisClusterAsyncContext *acc = redisClusterAsyncContextInit();
+
+// Add a cluster node address for the initial connect.
+redisClusterSetOptionAddNodes(acc->cc, "127.0.0.1:6379");
+
+// Attach an event engine. In this example we use libevent.
+struct event_base *base = event_base_new();
+redisClusterLibeventAttach(acc, base);
+
+if (redisClusterAsyncConnect2(acc) != REDIS_OK) {
+    printf("error: %s\n", acc->errstr);
+    exit(1);
 }
 ```
 
+#### Events per connection
+
```

Comment:
Copy-paste error? This heading is also on line 231.

Reply:
This header is for the async-section which has the same structure as sync-section above.
I though it was nice to have the same headers for similar information, ie:
```
## Cluster synchronous API
### Connecting
#### Events per connection                       (line 231)
   int redisClusterSetConnectCallback(...

## Cluster asynchronous API
### Connecting
#### Events per connection
  int redisClusterAsyncSetConnectCallback(..
```

Reply:
Ah, good idea. I hope the link targets still work though with two headings with the same name.

Should we mention how to use global events here too then? I.e.

> #### Events per cluster context
>
> Use `redisClusterSetEventCallback` with `acc->cc` for as the context.

Reply:
Seems to work fine with linkage when using same name headers, but there is a need to add `-1` to link to the second one like
```
[First](#events-per-cluster-context)
[Second](#events-per-cluster-context-1)
```

Reply:
Added the short version with a linkage for more details.